app/components/dashboard/run/left-panel/component.js

```diff
@@ -110,12 +112,13 @@ export default Component.extend(FullScreenMixin, {
 
     didInsertElement() {
         scheduleOnce('afterRender', this, () => {
-            // Check if we're coming from an ORCID redirect
-            // If ?auth=true
-            // Open Modal
-            const modalDialogName = 'ui/files/republish-modal';
-            this.showModal(modalDialogName, this.get('modalContext'));
-        
+          // Check if we're coming from an ORCID redirect
+          let queryParams = this.get('params')
+          if (queryParams) {
+            if (queryParams.auth === 'true') {
+              this.send('openPublishModal', this.model.taleId)
```

Comment:
```suggestion
              this.router.transitionTo({ queryParams: { auth: 'false' }});
              this.send('openPublishModal', this.model.taleId);
```

Reply:
For the above to work, I think line 17 (above) in this file also needs to change from
```javascript
    router: service('-routing')
```
to
```javascript
    router: service()
```

Reply:
One other thing to note, setting `auth` to `null` here will actually clear the query string parameter from the URL - this might be a better pattern than setting it to explicitly to `false`.

Reply:
Hmm I haven't been able to get this to work. The closest thing I saw was doing from [inside the route](https://github.com/emberjs/ember.js/blob/v3.9.1/packages/%40ember/-internals/routing/lib/system/route.ts#L661). 

I put this logic in the route, and also couldn't get it to reset the queryParams? I tried it with transitionTo and by manually setting them via this.set and didn't have any luck there.

Reply:
Hmmm... that's definitely strange.

All I needed was the following, and it appeared to work (in Chrome, at least):
```javascript
    router: service(),      // note: this is different from service('-routing')

    didInsertElement() {
        scheduleOnce('afterRender', this, () => {

            // Check if we're coming from an ORCID redirect
            let queryParams = this.get('params')
            if (queryParams && queryParams.auth === 'true') {
                this.router.transitionTo({ queryParams: { auth: null }});
                this.send('openPublishModal', this.model.taleId);
            }
        });
    },
```

You said that this doesn't clear the `auth` query string param for you?